src/screens/Feed/components/MediaModal/component.js

```diff
@@ -0,0 +1,178 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import algolia from 'algoliasearch/reactnative';
+import { kitsuConfig } from 'kitsu/config/env';
+import { View, Modal, FlatList, Keyboard, TouchableHighlight } from 'react-native';
+import { ModalHeader } from 'kitsu/screens/Feed/components/ModalHeader';
+import { SearchBox } from 'kitsu/components/SearchBox';
+import { isNull, upperFirst } from 'lodash';
+import { ProgressiveImage } from 'kitsu/components/ProgressiveImage';
+import { StyledText } from 'kitsu/components/StyledText';
+import * as Layout from 'kitsu/screens/Feed/components/Layout';
+import Icon from 'react-native-vector-icons/Ionicons';
+import { styles } from './styles';
+
+class MediaModal extends PureComponent {
+  static propTypes = {
+    visible: PropTypes.bool,
+    onCancelPress: PropTypes.func,
+    onMediaSelect: PropTypes.func,
+    algoliaKeys: PropTypes.object.isRequired,
+  }
+
+  static defaultProps = {
+    visible: false,
+    onCancelPress: null,
+    onMediaSelect: null,
+  }
+
+  state = {
+    media: [],
+    query: '',
+    page: 0,
+    loading: false,
+    selected: null,
+    apiKey: this.props.algoliaKeys.media.key,
+    indexName: this.props.algoliaKeys.media.index,
+  }
+
+  componentDidMount() {
+    this.doSearch(this.state.query, this.state.page);
+  }
+
+  doSearch = (query, page) => {
+    const { apiKey, indexName } = this.state;
+    const algoliaClient = algolia(kitsuConfig.algoliaAppId, apiKey);
+    const algoliaIndex = algoliaClient.initIndex(indexName);
+
+    algoliaIndex.setSettings({
+      attributesToRetrieve: [
+        'id',
+        'slug',
+        'kind',
+        'canonicalTitle',
+        'titles',
+        'posterImage',
+        'subtype',
+        'chapterCount',
+        'episodeCount',
+        'synopsis',
+      ],
+    });
+
+    this.setState({ loading: true });
+
+    algoliaIndex.search({ query, page }, (err, content) => {
+      if (!err) {
+        const media = page > 0 ? [...this.state.media, ...content.hits] : content.hits;
+        this.setState({ media });
+      }
+
+      this.setState({ loading: false });
```

Comment:
Not sure about this one, I mean, if there are no errors, you'll render the whole screen twice. What about this?

 ```
if (!err) {
    const media = page > 0 ? [...this.state.media, ...content.hits] : content.hits;
    this.setState({ media, loading: false });
} else {
    this.setState({ loading: false });
}
```

or we can just leave it as is